cognite/neat/_graph/extractors/_classic_cdf/_base.py

```diff
@@ -104,9 +116,48 @@ def extract(self) -> Iterable[Triple]:
             if self.limit and no >= self.limit:
                 break
 
-    @abstractmethod
     def _item2triples(self, item: T_CogniteResource) -> list[Triple]:
-        raise NotImplementedError()
+        id_value: str | None
+        if hasattr(item, "id"):
+            id_value = str(item.id)
+        else:
+            id_value = self._fallback_id(item)
+        if id_value is None:
+            return []
+
+        id_ = self.namespace[f"{self._instance_id_prefix}{id_value}"]
+
+        type_ = self._get_rdf_type(item)
+
+        # Set rdf type
+        triples: list[Triple] = [(id_, RDF.type, self.namespace[type_])]
+        if self.as_write:
+            item = item.as_write()
+        dumped = item.dump(self.camel_case)
+        dumped.pop("id", None)
+        # We have parentId so we don't need parentExternalId
+        dumped.pop("parentExternalId", None)
+        if "metadata" in dumped:
+            triples.extend(self._metadata_to_triples(id_, dumped.pop("metadata")))
+        if "columns" in dumped:
+            columns = dumped.pop("columns")
+            triples.append(
+                (id_, self.namespace.columns, Literal(json.dumps({"columns": columns}), datatype=XSD._NS["json"]))
+            )
+
+        for key, value in dumped.items():
```

Comment:
here we are sure that value is not list of values

Reply:
Yes that is handled in two lines below.